core/server/data/migrations/versions/4.43/2022-04-08-11-54-add-cancelled-events.js

```diff
@@ -0,0 +1,50 @@
+const ObjectID = require('bson-objectid').default;
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const cancelledSubscriptions = await knex
+            .select(
+                'members.id as member_id',
+                'members_stripe_customers_subscriptions.id',
+                'members_stripe_customers_subscriptions.plan_currency',
+                'members_stripe_customers_subscriptions.updated_at'
+            )
+            .from('members_stripe_customers_subscriptions')
+            .join('members_stripe_customers', 'members_stripe_customers.customer_id', '=', 'members_stripe_customers_subscriptions.customer_id')
+            .join('members', 'members_stripe_customers.member_id', '=', 'members.id')
+            .where('members_stripe_customers_subscriptions.cancel_at_period_end', true)
+            .whereNot('members_stripe_customers_subscriptions.status', 'canceled');
+
+        if (cancelledSubscriptions.length === 0) {
+            logging.info('No missing cancelled events - skipping migration');
+            return;
+        }
+
+        const eventsToInsert = cancelledSubscriptions.map((subscription) => {
+            const event = {
+                id: (new ObjectID()).toHexString(),
+                type: 'canceled',
+                member_id: subscription.member_id,
+                subscription_id: subscription.id,
+                from_plan: null,
+                to_plan: null,
```

Comment:
Rethinking this: will we be able to answer the question later on of "what plan were they on when they cancelled?" 🤔

Reply:
🤔 Hmm - no - I guess we should store it as `from_plan` and `to_plan` ?

Reply:
Yeah I think that makes sense

Reply:
Updated!